src/main/util.ts

```diff
@@ -247,74 +259,98 @@ const getVideoState = (storageDir: unknown) => {
 }    
 
 /**
- *  writeMetadataFile
+ *  Writes video metadata asynchronously and returns a Promise
  */
-const writeMetadataFile = (storageDir: string, metadata: Metadata) => {
+const writeMetadataFile = async (storageDir: string, metadata: Metadata) => {
+    const file = await getNewestVideo(storageDir);
     const jsonString = JSON.stringify(metadata, null, 2);
-    const newestVideoPath = getNewestVideo(storageDir);
-    const newestVideoName = path.basename(newestVideoPath, '.mp4');
-    fs.writeFileSync(storageDir + newestVideoName + ".json", jsonString);
+    const metadataFileName = getMetadataFileForVideo(file);
+
+    return await fs.writeFile(metadataFileName, jsonString);
 }    
 
 /**
- * runSizeMonitor, maxStorage in GB.
+ * Return information about a video needed for various parts of the application
  */
-const runSizeMonitor = (storageDir: any, maxStorageGB: any) => {
-    console.debug("Running size monitor");  
-    const maxStorageBytes = maxStorageGB * Math.pow(1024, 3);
-    let totalSize = 0;
-    const files = fs.readdirSync(storageDir);
+const getVideoInfo = (videoPath: string): VideoInfo => {
+    const fstats = fscb.statSync(videoPath);
+    const mtime = fstats.mtime;
+    const size = fstats.size;
 
-    for (const file of files) {
-        totalSize += fs.statSync(storageDir + file).size;
-    }
+    return { name: videoPath, size, mtime };
+};
 
-    if (totalSize > maxStorageBytes) { 
-        deleteOldestVideo(storageDir);
-        runSizeMonitor(storageDir, maxStorageBytes);
-    } 
-}   
+const getSortedVideos = async (storageDir: string): Promise<VideoInfo[]> => {
```

Comment:
Worth a comment to say which sorted order this is, can work it out  from the func but it's a not totally obvious.